.github/workflows/pr.yaml

```diff
@@ -47,21 +39,69 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v4
         with:
           go-version: 1.18
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/go/pkg/mod
-            ~/.cache/go-build
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
       - name: run tests
         run: go test -timeout 30m -json ./... > test.json
       - name: Annotate tests
         if: always()
         uses: guyarb/golang-test-annotations@v0.5.0
         with:
           test-results: test.json
+
+  generate-client:
+    runs-on: ubuntu-latest
+    env:
+      TESTNET_ACCOUNT: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+      MAINNET_FIND: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+      MAINNET_FIND_ADMIN: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+    needs: test
+    output:
+      client_changed: ${{ steps.bump.outputs.bumped }}
+    steps:
+     - name: Checkout code
+       uses: actions/checkout@v3
+     - uses: actions/setup-go@v4
+       with:
+         go-version: 1.18
+     - run: make client
+     - name: verify client changed
+       uses: tj-actions/verify-changed-files@v14
+       id: verify-changed-files
+       with:
+         files: |
+           lib/find.json
+           dapper-tx
+      - name: fail if dapper transactions changed
+       if: contains(steps.verify-changed-files.outputs.changed_files, 'dapper-tx')
+       run: |
+        echo "dapper-tx is not updated"
+        exit 1
+     - name: bump client if find client changed
```

Comment:
if client is changed and package.json is changed we dont have to bump as long as the version is rc.